src/app/delegation_backend/src/itn_uptime_analyzer/sheets.go

```diff
@@ -0,0 +1,164 @@
+	readRange := ITN_UPTIME_ANALYZER_SHEET + "!" + col + ":" + col
+	spId := OutputSpreadsheetId()
+
+	identityString := strings.Join([]string{identity["public-key"], identity["public-ip"]}, "-")
+
+	cellValue := []interface{}{identityString}
+
+	valueRange := sheets.ValueRange{
+		Values: [][]interface{}{cellValue},
+	}
+
+	_, err := client.Spreadsheets.Values.Append(spId, readRange, &valueRange).ValueInputOption("USER_ENTERED").Do()
+	if err != nil {
+		log.Fatalf("Unable to append data to sheet: %v", err)
+	}
+}
+
+// Inserts the identity string of the node in the first column under rowIndex
+
+func (identity Identity) InsertBelow(client *sheets.Service, log *logging.ZapEventLogger, rowIndex int) {
+	col := IDENTITY_COLUMN
+	readRange := fmt.Sprintf("%s!%s%d:%s%d", ITN_UPTIME_ANALYZER_SHEET, col, rowIndex + 1, col, rowIndex + 1)
+	spId := OutputSpreadsheetId()
+
+	identityString := strings.Join([]string{identity["public-key"], identity["public-ip"]}, "-")
+
+	cellValue := []interface{}{identityString}
+
+	valueRange := sheets.ValueRange{
+		Values: [][]interface{}{cellValue},
+	}
+
+	_, err := client.Spreadsheets.Values.Append(spId, readRange, &valueRange).ValueInputOption("USER_ENTERED").InsertDataOption("INSERT_ROWS").Do()
+	if err != nil {
+		log.Fatalf("Unable to insert data in sheet: %v", err)
+	}
+}
+
+// Finds the first empty column on the row specified and puts up or not up
+
+func (identity Identity) AppendUptime(client *sheets.Service, log *logging.ZapEventLogger, rowIndex int) {
+	readRange := fmt.Sprintf("%s!A%d:Z%d", ITN_UPTIME_ANALYZER_SHEET, 1, 1)
+	spId := OutputSpreadsheetId()
+
+	resp, err := client.Spreadsheets.Values.Get(spId, readRange).Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v\n", err)
+	}
+
+	var nextEmptyColumn int = len(resp.Values[0])
+
+	updateRange := fmt.Sprintf("%s!%s%d", ITN_UPTIME_ANALYZER_SHEET, string(nextEmptyColumn + 65), rowIndex)
+
+	var cellValue []interface{}
+
+	uptimeArrayLength, err := strconv.Atoi(identity["uptime"])
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v\n", err)
+	}
+
+	if (uptimeArrayLength >= 47) {
+		cellValue = []interface{}{"up"}
+	} else {
+		cellValue = []interface{}{"not up"}
+	}
+
+	valueRange := sheets.ValueRange{
+		Values: [][]interface{}{cellValue},
+	}
+
+	_, err = client.Spreadsheets.Values.Append(spId, updateRange, &valueRange).ValueInputOption("USER_ENTERED").Do()
+	if err != nil {
+		log.Fatalf("Unable to insert data in sheet: %v\n", err)
+	}
+}
+
+// Tracks the date of execution on the top row of the spreadsheet
+
+func MarkExecution(client *sheets.Service, log *logging.ZapEventLogger){
+	readRange := fmt.Sprintf("%s!A%d:Z%d", ITN_UPTIME_ANALYZER_SHEET, 1, 1)
+	spId := OutputSpreadsheetId()
+
+	currentTime := GetCurrentTime()
+	lastExecutionTime := GetLastExecutionTime(currentTime)
+
+	timeInterval := strings.Join([]string{currentTime.Format(time.RFC3339), lastExecutionTime.Format(time.RFC3339)}, " - ")
```

Comment:
Time formatting implicitly relies on the host server to be configured for UTC. This is not necessary the case and a robist solution would look like: `currentTime.UTC().Format(time.RFC3339)` (to be used anywhere where we convert time to string to be used for google bucket)